mempool/reactor.go

```diff
@@ -163,20 +161,15 @@ func (memR *Reactor) Receive(chID byte, src p2p.Peer, msgBytes []byte) {
 	}
 	memR.Logger.Debug("Receive", "src", src, "chId", chID, "msg", msg)
 
-	switch msg := msg.(type) {
-	case *TxMessage:
-		txInfo := TxInfo{SenderID: memR.ids.GetForPeer(src)}
-		if src != nil {
-			txInfo.SenderP2PID = src.ID()
-		}
-		err := memR.mempool.CheckTx(msg.Tx, nil, txInfo)
-		if err != nil {
-			memR.Logger.Info("Could not check tx", "tx", txID(msg.Tx), "err", err)
-		}
-		// broadcasting happens from go routines per peer
-	default:
-		memR.Logger.Error(fmt.Sprintf("Unknown message type %v", reflect.TypeOf(msg)))
+	txInfo := TxInfo{SenderID: memR.ids.GetForPeer(src)}
```

Comment:
won't this crash without the switch if I send something other than `TxMessage` to mempool reactor?

Reply:
well now the msg is not an interface like it was before. Before we were returning a msg: https://github.com/tendermint/tendermint/pull/4409/files#diff-faa5920fe04b5bb0d81cec8ad247ccdcL268 which was an unneeded interface so if it is not able to decode the message then it would panic in the previous times. https://github.com/tendermint/tendermint/pull/4409/files#diff-faa5920fe04b5bb0d81cec8ad247ccdcL268

I can add a switch statement but since there is only 1 msg type then it seems a bit redundant

Reply:
> since there is only 1 msg type then it seems a bit redundant

Agree

Reply:
also panic would accure here: https://github.com/tendermint/tendermint/pull/4409/files#diff-faa5920fe04b5bb0d81cec8ad247ccdcR156 if it was not of type txmessage